map/src/menu/tracks/actions/TracksHeader.jsx

```diff
@@ -0,0 +1,169 @@
+import React, { useContext, useState } from 'react';
+import AppContext from '../../../context/AppContext';
+import { ReactComponent as CloseIcon } from '../../../assets/icons/ic_action_close.svg';
+import { ReactComponent as BackIcon } from '../../../assets/icons/ic_arrow_back.svg';
+import { ReactComponent as ImportIcon } from '../../../assets/icons/ic_action_folder_import_outlined.svg';
+import { ReactComponent as AddFolderIcon } from '../../../assets/icons/ic_action_folder_add_outlined.svg';
+import { MENU_INFO_CLOSE_SIZE } from '../../../manager/GlobalManager';
+import { Dialog } from '@material-ui/core';
+import DialogActions from '@mui/material/DialogActions';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogContent from '@mui/material/DialogContent';
+import styles from '../trackmenu.module.css';
+import dialogStyles from '../../../dialogs/dialog.module.css';
+import { prepareFileName } from '../../../util/Utils';
+import { saveEmptyTrack } from '../../../manager/track/SaveTrackManager';
+import { DEFAULT_GROUP_NAME } from '../../../manager/track/TracksManager';
+
+export default function TracksHeader({ trackGroup, sortIcon, setOpenSort, anchorEl }) {
+    const ctx = useContext(AppContext);
+
+    const [openAddFolderDialog, setOpenAddFolderDialog] = useState(false);
+
+    function closeTrackMenu() {
+        ctx.setInfoBlockWidth(MENU_INFO_CLOSE_SIZE);
+    }
+
+    function prevTrackMenu() {
+        ctx.openTrackGroups.pop();
+        ctx.setOpenTrackGroups([...ctx.openTrackGroups]);
+    }
+
+    function isFolderExist(name) {
+        return trackGroup.subfolders.some((folder) => folder.name === name);
+    }
+
+    const DialogAddFolder = () => {
+        const [folderNameError, setFolderNameError] = useState('');
+        const [folderName, setFolderName] = useState('');
+
+        function validationFolderName(name) {
+            if (!name || name === '' || name.trim().length === 0) {
+                setFolderNameError('Empty folder name!');
+            } else if (isFolderExist(name)) {
+                setFolderNameError('Folder with same name already exist!');
+            } else {
+                setFolderNameError('');
+            }
+        }
+
+        async function addFolder() {
+            let folderPart;
+            if (trackGroup.fullName === DEFAULT_GROUP_NAME) {
+                folderPart = '';
+            } else {
+                folderPart = `${trackGroup.fullName}/`;
+            }
+            const newFolderName = `${folderPart}${folderName}`;
+            saveEmptyTrack(newFolderName, ctx).then();
+            setOpenAddFolderDialog(false);
+        }
+
+        return (
+            <Dialog open={true} onClose={() => setOpenAddFolderDialog(false)}>
+                <DialogTitle className={dialogStyles.title}>Add new folder</DialogTitle>
+                <DialogContent className={dialogStyles.content}>
+                    <TextField
```

Comment:
1) please run `addFolder()` by Enter key with onKeyPress handler

2) please use `label` or remove unused empty space on the top of the input field